tests/Unit/Utilities/CollectionExtensionsTests.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Gossip.Utilities;
+using NUnit.Framework;
+
+namespace Gossip.UnitTests.Utilities
+{
+    [TestFixture]
+    public class CollectionExtensionsTests
+    {
+        private IEnumerable<int> _list;
+        [SetUp]
+        public void SetUp()
+        {
+            _list = new List<int> { 1, 2, 3, 4, 5, 6 };
+        }
+
+        [Test]
+        public void Batch_throws_an_error_when_batch_size_is_zero()
+        {
+            // arrange
+            var wasExceptionCaught = false;
+            var errorMessage = string.Empty;
+            
+            // act
+            try
+            {
+                foreach (var batch in _list.Batch(0)) { }
+            }
+            catch(Exception e)
+            {
+                errorMessage = e.Message;
+                wasExceptionCaught = true;
+            }
+            
+            // assert
+            Assert.IsTrue(wasExceptionCaught);
+            Assert.AreEqual(errorMessage, "Batch size cannot be zero unless you like endless CPU cycles!");
+        }
+        
+        [Test]
+        public void Batch_creates_a_batch()
+        {
+            // act
+            foreach (var batch in _list.Batch(2))
+            {
+                // assert
+                Assert.AreEqual(batch.Length, 2);
```

Comment:
Do we want to assert anything about the contents of these batches? Or just their length?

Reply:
I was originally thinking of this, however with looking at the method, .Batch doesn't really care about the contents so I decided to move forward from that. The method `.BatchNoMultipleEnumeration` is more appropriate to test what the contents are since it takes in a function to apply to the batch. I went ahead and did a simple test on that as well.

Reply:
Fair. I'm asking "do we want to test that the right things are in each batch?" not because the method *should* be manipulating the contents, but because when you batch the list you do have an expectation that the resulting contents of each batch will be predictable. Example: [a, b, c, d].Batch(2) should get you [a, b] and [c, d] but not [a, b] [a, b]. Testing just for batch length doesn't verify this. Not required, just another assertion you could make if you wanted. :+1: